libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -978,8 +980,30 @@ void IRGeneratorForStatements::endVisit(IndexAccess const& _indexAccess)
 			}
 			case DataLocation::CallData:
 			{
-				solUnimplemented("calldata not yet implemented!");
-
+				vector<string> vars { m_context.newYulVariable() };
+				if (arrayType.baseType()->isDynamicallySized())
+					vars.emplace_back(m_context.newYulVariable());
+				m_code << "let " <<
+					joinHumanReadable(vars) <<
+					" := " <<
+					m_utils.calldataArrayIndexAccessFunction(arrayType) <<
+					"(" <<
+					m_context.variable(_indexAccess.baseExpression()) <<
+					", " <<
+					expressionAsType(*_indexAccess.indexExpression(), *TypeProvider::uint256()) <<
+					")\n";
+				if (arrayType.isByteArray())
+					defineExpression(_indexAccess) << m_utils.conversionFunction(*arrayType.baseType(), *arrayType.baseType()) << "(calldataload(" << vars.front() << "))\n";
```

Comment:
Does it perform cleanup?

Reply:
``m_utils.conversionFunction(*arrayType.baseType(), *arrayType.baseType())`` does nothing but cleanup - at least that was the idea, but I'll double-check.